mkosi

```diff
@@ -557,6 +557,17 @@ def mount_image(args, workspace, loopdev, root_dev, home_dev, srv_dev, root_read
 
             umount(root)
 
+@complete_step("Assigning hostname")
+def assign_hostname(args, workspace):
+    if not args.hostname:
+        return
+
+    root = os.path.join(workspace, "root")
+    hostname_path = os.path.join(root, "etc/hostname")
+    
+    with open(hostname_path, "w+") as f:
```

Comment:
Debootstrap will try to copy `/etc/hostname` from the host, so this affects debian-based distros too.